snuba/migrations/runner.py

```diff
@@ -57,6 +57,44 @@ def __init__(self) -> None:
             ClickhouseClientSettings.MIGRATE
         )
 
+        self.__status: MutableMapping[
+            MigrationKey, Tuple[Status, Optional[datetime]]
+        ] = {}
+
+    def get_status(
+        self, migration_key: MigrationKey
+    ) -> Tuple[Status, Optional[datetime]]:
+        """
+        Returns the status and timestamp of a migration.
+        """
+
+        if self.__status.get(migration_key):
+            return self.__status[migration_key]
+
+        try:
+            data = self.__connection.execute(
+                f"SELECT status, timestamp FROM {self.__table_name} FINAL WHERE group = %(group)s AND migration_id = %(migration_id)s",
+                {
+                    "group": migration_key.group.value,
+                    "migration_id": migration_key.migration_id,
+                },
+            )
+
+            if data:
+                status, timestamp = data[0]
```

Comment:
I'm not super familiar with the migration code, is it possible to have more than one result here?

Reply:
It's not because the primary key on this table is `group, migration_id` - duplicate rows would be merged together so there's only ever one row representing a single migration.